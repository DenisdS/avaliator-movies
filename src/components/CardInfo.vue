<script>
export default {
  name: 'CardInfo',
  props: {
    card: {
      type: Object,
      required: true,
      default: () => ({})
    }
  }
}
</script>

<template>
  <div>
    CardInfo
  </div>
</template>