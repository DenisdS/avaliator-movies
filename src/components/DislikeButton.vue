<script>
export default {
  name: 'DislikeButton'
}
</script>

<template>
  <button>
    Dislike
  </button>
</template>