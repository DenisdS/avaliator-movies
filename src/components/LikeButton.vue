<script>
export default {
  name: 'LikeButton',
  data() {
    return {
      title: ''
    }
  }
}
</script>

<template>
  <form @click.prevent="$emit('liked', title)">
    <input type="text" v-model="title" />
    <button type="button">Enviar</button>
  </form>
</template>