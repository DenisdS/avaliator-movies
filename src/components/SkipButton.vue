<script>
export default {
  name: 'SkipButton'
}
</script>

<template>
  <div>
    Skip
  </div>
</template>