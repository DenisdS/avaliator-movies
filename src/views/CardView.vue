<script>
import CardInfo from '@/components/CardInfo'
import LikeButton from '@/components/LikeButton'
import SkipButton from '@/components/SkipButton'
import DislikeButton from '@/components/DislikeButton'

export default {
  name: 'CardView',
  components: {
    CardInfo,
    LikeButton,
    SkipButton,
    DislikeButton
  },
  data () {
    return {
      card: { movie: '' } 
    }
  }
}
</script>

<template>
  <div>
    <CardInfo :card="card" />
    <LikeButton />
    <SkipButton />
    <DislikeButton />
  </div>
</template>
