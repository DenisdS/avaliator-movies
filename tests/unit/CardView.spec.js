import { shallowMount } from '@vue/test-utils'
import CardView from '@/views/CardView'
import LikeButton from '@/components/LikeButton.vue'
import DislikeButton from '@/components/DislikeButton.vue'
import SkipButton from '@/components/SkipButton.vue'
import CardInfo from '@/components/CardInfo.vue'

describe('CardView', () => {
  const build = () => {
    const wrapper = shallowMount(CardView, {
      data: () => ({
        card: {}
      })
    })

    return {
      wrapper,
      likeButton: () => wrapper.find(LikeButton),
      dislikeButton: () => wrapper.find(DislikeButton),
      skipButton: () => wrapper.find(SkipButton),
      cardInfo: () => wrapper.find(CardInfo)
    }
  }
  
  it('renders the component', () => {
    const { wrapper } = build()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders main child components', () => {
    const { likeButton, skipButton, dislikeButton, cardInfo } = build()

    expect(likeButton().exists()).toBe(true)
    expect(skipButton().exists()).toBe(true)
    expect(dislikeButton().exists()).toBe(true)
    expect(cardInfo().exists()).toBe(true)
  })

  it('passes a binded of the infos of card', () => {
    const { wrapper, cardInfo } = build()
    wrapper.setData({
      card: {
        movie: 'Matrix'
      }
    })

    expect(cardInfo().vm.card).toBe(wrapper.vm.card)
  })
})


