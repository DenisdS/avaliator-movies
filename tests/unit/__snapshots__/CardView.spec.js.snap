// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardView renders the component 1`] = `
<div>
  <cardinfo-stub card="[object Object]"></cardinfo-stub>
  <likebutton-stub></likebutton-stub>
  <skipbutton-stub></skipbutton-stub>
  <dislikebutton-stub></dislikebutton-stub>
</div>
`;
